### PR TITLE
Pointer to syntax used for format

### PR DESCRIPTION
**Pull request by mpergler**
Added description of what syntax/parsing engine is used for the formatting. Motivated by #42806, but in general a good idea so users can look up more in detail what is possible. Also uniformized spelling (placeholder vs placeholder).

### CODE REVIEW

File: resources/function_help/json/format [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "groups": ["String"],
   "description": "Format a string using supplied arguments.",
   "variableLenArguments": true,
-  "arguments": [ {"arg":"string", "description":"A string with place holders for the arguments. Use %1, %2, etc for placeholders. Placeholders can be repeated."},
+  "arguments": [ {"arg":"string", "description":"A string with placeholders %1, %2, etc., for the arguments. Placeholders can be repeated. Follows the syntax of Qt's QString::arg."},
```
Comment (DelazJ) [inline resources/function_help/json/format:7]: @Houska1 what about adding the url to that at feature, if any? Could skip one more web search for users.

Comment (m-kuhn) [reply resources/function_help/json/format:7]: For reference, here's how the qt docs express it:

Returns a copy of this string with the lowest numbered place marker replaced by string a, i.e., %1, %2, ..., %99.

Comment (Houska1) [reply resources/function_help/json/format]: Great idea on the url @DelazJ ; added (please check my escaping \" in the href -- can't compile to test). 
Thanks for the reference @m-kuhn. I used the link but didn't put in the "lowest numbered place marker" detail -- seemed too detailed for brief help, and me tripping over it seemed an edge case. More important to let users know where they can get more info how it works overall.

Comment (DelazJ) [reply resources/function_help/json/format]: @Houska1 I think the escaping is correct. You could also have used the single quotes to avoid escaping hassle. see the overlay_* or uuid functions
If I may, avoid `@`mention people in commit messages, otherwise that person get notifications for every person that updates their branch with your commits. I can certify it's quite noisy. 😄

Comment (nyalldawson) [reply resources/function_help/json/format]: I'd suggest copying the wording @m-kuhn has provided and skipping the link to Qt's docs (better to have everything right in front of the user instead of forcing them to seek and discover this!)

Comment (Houska1) [reply resources/function_help/json/format]: OK, added. Also added a 2nd example to illustrate, i.e. not just the obvious ... %1 ... %2 ... behaving as expected.

File: resources/function_help/json/format [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
+  "description": "Format a string using supplied arguments. Based on <a href=\"https://doc.qt.io/qt-5/qstring.html#arg\">Qt's QString::arg</a>.",
   "variableLenArguments": true,
-  "arguments": [ {"arg":"string", "description":"A string with place holders for the arguments. Use %1, %2, etc for placeholders. Placeholders can be repeated."},
+  "arguments": [ {"arg":"string", "description":"A string with placeholders %1, %2, etc., for the arguments. Placeholders can be repeated. The lowest numbered placeholder is replaced by arg1, the next by arg2, etc."},
                  {"arg":"arg1", "syntaxOnly": true},
                  {"arg":"arg2", "syntaxOnly": true},
                  {"arg":"arg", "descOnly": true, "description":"any type. Any number of arguments."}],
-  "examples": [ { "expression":"format('This %1 a %2','is', 'test')", "returns":"'This is a test'"}
-  ]
+  "examples": [ { "expression":"format('This %2 a %1','test', 'is')", "returns":"'This is a test'"},
```
Comment (m-kuhn) [inline resources/function_help/json/format:11]: Reverse ordering is an advanced use case. For readability I would cover the standard first.

File: resources/function_help/json/format (hunk 1 above)
Comment (m-kuhn) [reply resources/function_help/json/format]: Should we also remove the link? It's embedded in much irrelevant info specific for qt/c++

Comment (nyalldawson) [reply resources/function_help/json/format]: I think so. I dislike pointing users to the c++ documentation -- it's needlessly complex for them and likely to be incredibly intimidating to the bulk of our users.

Comment (Houska1) [reply resources/function_help/json/format]: Happy to revert to the first exmple to be plain vanilla. (Will do so after we resolve the below)

However, I'd push back a bit on the suggestion to remove the link. I'm not going to die on my sword to specifically have it here, but our documentation approach overall has a bit of a flaw. Embedded help and on-line docs taken together tend to cover basic use cases, but are silent on nuances and are silent on _other sources of information people can look to if they wish_. The best documentation tends to peel the onion, accurately giving the basics, then giving more where appropriate, but also breadcrumbs if even more detail is needed. 

While it's not necessarily high priority to go and add all of this, and this little function is small potatoes, when the opportunity presents to add such breadcrumbs, why not take it? With that in mind, given it's come up, if the algorithm here is based on something else which is well documented, why not give users the pointer? 

You've encouraged me to add specifics here, which was good. And maybe further breadcrumbs, if any, belong in the on-line manual rather than here. But I'd push back against "don't leave breadcrumbs since they lead to stuff that's hard".
I'd also suspect most people writing more complex QGIS expressions have some coding experience, whether its C++, Python, R, VBA, heck Fortran77, so I'm not sure developer-focused docs are necessarily off-putting in expression functions in particular even if they might be for GUI-only QGIS use.

Anyway, off my soapbox. This one quirk of one function isn't worth this many words, and if you say "whatever, but just take the link out", I will :)

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Thanks @Houska1 !
